api/python/quilt3/packages.py

```diff
@@ -865,35 +863,35 @@ def build(self, name=None, registry=None, message=None):
         Returns:
             The top hash as a string.
         """
-        return self._build(name=name, registry=registry, message=message)
-
-
-    def _build(self, name=None, registry=None, message=None):
-        self._set_commit_message(message)
+        validate_package_name(name)
```

Comment:
This should be a part of `_build`, not a part of `build`

Reply:
I'm kind of on the fence here: I feel like user-facing APIs should take user-friendly arguments and do validation - while internal APIs should take whatever is convenient for us (e.g., `PhysicalKey` rather than a `str`) and not bother with validation cause it's unnecessary.

`validate_package_name` is pretty harmless - but passing the registry as a string and parsing it in the caller and in the callee is kind of silly.

In case of `_build` in particular, it doesn't really matter since it's only called by `build`. I'm just wondering what the convention should be in general. Maybe using different names for parameters or type annotations would help.

Reply:
Right now `build` should just be `_build` + telemetry. It's not great having two methods that do the same thing, but I think we can reduce complexity if we always know that `_method` does the exact same thing as `method` but without the telemetry. 

Longer term it would really be nice to have that distinction between user-facing APIs and internal APIs. and split up the logic accordingly, but I think we probably want to restructure the code into multiple classes for that (to make it very clear to users what APIs are public and reliable and what APIs are private and prone to breaking changes).